domain/infosync/resource_group_manager.go

```diff
@@ -0,0 +1,218 @@
+	}
+	var isRUMode bool
+	if modelGroup.RRURate > 0 || modelGroup.WRURate > 0 {
+		isRUMode = true
+		group.Settings.Mode = rmpb.GroupMode_RUMode
+		group.Settings.RUSettings = &rmpb.GroupRequestUnitSettings{
+			RRU: &rmpb.TokenBucket{
+				Settings: &rmpb.TokenLimitSettings{
+					Fillrate: modelGroup.RRURate,
+				},
+			},
+			WRU: &rmpb.TokenBucket{
+				Settings: &rmpb.TokenLimitSettings{
+					Fillrate: modelGroup.WRURate,
+				},
+			},
+		}
+	}
+	if len(modelGroup.CPULimiter) > 0 || len(modelGroup.IOReadBandwidth) > 0 || len(modelGroup.IOWriteBandwidth) > 0 {
+		if isRUMode {
+			return nil, errors.New("Only one of RUMode and NativeMode can be set")
+		}
+		parseF := func(s string, scale resource.Scale) (uint64, error) {
+			if len(s) == 0 {
+				return 0, nil
+			}
+			q, err := resource.ParseQuantity(s)
+			if err != nil {
+				return 0, err
+			}
+			return uint64(q.ScaledValue(scale)), nil
+		}
+		cpuRate, err := parseF(modelGroup.CPULimiter, resource.Milli)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		ioReadRate, err := parseF(modelGroup.IOReadBandwidth, resource.Scale(0))
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		ioWriteRate, err := parseF(modelGroup.IOWriteBandwidth, resource.Scale(0))
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+
+		group.Settings.Mode = rmpb.GroupMode_NativeMode
+		group.Settings.ResourceSettings = &rmpb.GroupResourceSettings{
+			Cpu: &rmpb.TokenBucket{
+				Settings: &rmpb.TokenLimitSettings{
+					Fillrate: cpuRate,
+				},
+			},
+			IoRead: &rmpb.TokenBucket{
+				Settings: &rmpb.TokenLimitSettings{
+					Fillrate: ioReadRate,
+				},
+			},
+			IoWrite: &rmpb.TokenBucket{
+				Settings: &rmpb.TokenLimitSettings{
+					Fillrate: ioWriteRate,
+				},
+			},
+		}
+		return group, nil
+	}
+	if isRUMode {
+		return group, nil
+	}
+	return nil, errors.New("at least one setting in the group")
+}
+
+// ResourceGroupManager manages resource group settings
+type ResourceGroupManager interface {
+	// GetResourceGroup is used to get one specific rule bundle from ResourceGroup Manager.
+	GetResourceGroup(ctx context.Context, name string) (*rmpb.ResourceGroup, error)
+	// GetAllResourceGroups is used to get all rule bundles from ResourceGroup Manager.
+	GetAllResourceGroups(ctx context.Context) ([]*rmpb.ResourceGroup, error)
+	// PutResourceGroup is used to post specific rule bundles to ResourceGroup Manager.
+	CreateResourceGroup(ctx context.Context, group *rmpb.ResourceGroup) error
+	// ModifyResourceGroup is used to modify specific rule bundles to ResourceGroup Manager.
+	ModifyResourceGroup(ctx context.Context, group *rmpb.ResourceGroup) error
+	// DeleteResourceGroup is used to delete specific rule bundles to ResourceGroup Manager.
+	DeleteResourceGroup(ctx context.Context, name string) error
+}
+
+// ExternalResourceGroupManager manages placement with resource manager.
+// TODO: replace with resource manager client.
+type ExternalResourceGroupManager struct {
```

Comment:
Is it necessary to expose the struct?
Or maybe 
```
func NewResourceGroupManager() ResourceGroupManager {
    return &externalResourceGroupManager{}
}
```

If you expose the `ExternalResourceGroupManager` struct, while the field `etcdCli` is not exposed, how to create a `ExternalResourceGroupManager` instance?